### ISSUE TITLE
[BUG] Badly-constructed GET request fired at an useForm / useModalForm with no need (and gets 404/500)

### ISSUE THREAD

**Issue by banuni** [pankod/refine#1260]
**Describe the bug**
When using `useForm` or `useModalForm` on resource A in the **Show** context of resource B - a GET request is being performed to resource A with the ID of shown resource B.
**To Reproduce**
1. have two resources `Person` and `Car`
2. Create a `ShowPerson` page with `useModalForm` to create a `Car` for that `Person`:  (note that all usage of the hook results is commented out, and this still happens)
```tsx
export const PersonShow = () => {
  const { queryResult } = useShow();
  const { data, isLoading } = queryResult;
  const record: any = data?.data;
  const { list } = useNavigation();

  const { modalProps, formProps, show } = useModalForm({
    action: 'create',
    resource: 'cars',
    defaultFormValues: { person_id: record?.id },
  });

  return (
    <>
      <Show isLoading={isLoading}>
        <Title level={5}>Person Name</Title>
        <Text>{record?.person.name}</Text>

        <Title level={5}>ID</Title>
        <Text>
          <Tag>{record?.id}</Tag>
        </Text>

        <Title level={5}>Actions</Title>
        <ButtonToolbar>
          <DeleteButton
            onSuccess={() => {
              list('persons');
            }}
          />
          {/* <Button onClick={() => show()}>Create Car</Button> */}
        </ButtonToolbar>
      </Show>
      {/* <Modal {...modalProps}>
        <Form {...formProps} layout="vertical">
          <Form.Item label="Person ID" name="person_id">
            <Input disabled/>
          </Form.Item>
          <Form.Item label="Car Model" name="model">
            <Input/>
          </Form.Item>
        </Form>
      </Modal> */}
    </>
  );
};
```

3. go to that *Show* page
4. See 404 popup (and 404 in browser network tab) - the request is done to `/api/cars/the_person_id`

**Expected behavior**
No request should be made for that hook...since this is a create form (with no `useOptions` etc...")
At most the request should be done to `/api/persons/the_person_id`

**Desktop:**
 - OS: MacOS 11.6 (Big Sur)
 - Browser: chrome
 - Version 95.0.4638.69
 - Used dataProvider `refine-simple-rest`

**Comment by omeraplak** [comment]
Hey @banuni ,
I have good and bad news for you.
Good;
We identified the source of the issue.
Bad,
The problem has multiple possible solutions. We should discuss this with the core team. Although I cannot give an exact date, I think it will be resolved within this week

**Comment by banuni** [comment]
Amazing! For now this doesn't block me since the only  sode effect is a popup of a 404 on the UI. Thanks for the response :)

**Comment by omeraplak** [comment]
Hey @banuni ,
We've released 2.2.2. Could you approve this issue is fixed?

**Comment by banuni** [comment]
Hey @omeraplak , sorry for the delayed response, it is indeed fixed! thank you!!!
